nixos/modules/services/web-apps/claper.nix

```diff
@@ -0,0 +1,133 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.claper;
+in {
+  options.services.claper = {
+    enable = mkEnableOption (lib.mdDoc "claper");
+
+    database = {
+      postgres = {
+        setup = mkEnableOption (lib.mdDoc "creating a postgresql instance") // { default = true; };
+        dbname = mkOption {
+          default = "claper";
+          type = types.str;
+          description = lib.mdDoc ''
+            Name of the database to use.
+          '';
+        };
+        url = mkOption {
+          default = "postgres://claper:claper@127.0.0.1:5432/claper";
+          type = types.str;
+          description = lib.mdDoc ''
+            Path to the UNIX domain-socket to communicate with `postgres`.
+          '';
+        };
+      };
+    };
+
+    server = {
+      secretKeybaseFile = mkOption {
+        type = types.either types.path types.str;
+        description = lib.mdDoc ''
+          Path to the secret used by the `phoenix`-framework. Instructions
+          how to generate one are documented in the
+          [
+          framework docs](https://hexdocs.pm/phoenix/Mix.Tasks.Phx.Gen.Secret.html#content).
+        '';
+      };
+      endpoint_host = mkOption {
+        default = "localhost";
+        type = types.str;
+        description = lib.mdDoc ''
+          Host used to access the app (used for url generation)
+        '';
+      };
+      endpoint_port = mkOption {
+        default = 80;
+        type = types.port;
+        description = lib.mdDoc ''
+          Port used to access the app (used for url generation)
+        '';
+      };
+      port = mkOption {
+        default = 4000;
+        type = types.port;
+        description = lib.mdDoc ''
+          Port to listen to
+        '';
+      };
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+    services.postgresql = mkIf cfg.database.postgres.setup {
+      enable = true;
+    };
+
+    services.epmd.enable = true;
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services = mkMerge [
+      {
+        claper = {
+          inherit (pkgs.claper.meta) description;
+          documentation = [ "https://docs.claper.co" ];
+          wantedBy = [ "multi-user.target" ];
+          after = optionals cfg.database.postgres.setup [
+            "postgresql.service"
+            "claper-postgres.service"
+          ];
+          requires = optionals cfg.database.postgres.setup [
+              "postgresql.service"
+              "claper-postgres.service"
+            ];
+
+          environment = {
+            DATABASE_URL = cfg.database.postgres.url;
+            SECRET_KEY_BASE = cfg.secret_key_base;
+            ENDPOINT_HOST = cfg.server.endpoint_host;
+            ENDPOINT_PORT = cfg.server.endpoint_port;
+            PORT = cfg.server.port;
```

Comment:
We should use socket authentication

Reply:
I tried to use socket but Claper requires an URL.